semantickernel-experimental/src/main/java/com/microsoft/semantickernel/connectors/data/jdbc/JDBCVectorStoreRecordCollection.java

```diff
@@ -0,0 +1,312 @@
+     * @param options the options
+     */
+    @SuppressFBWarnings("EI_EXPOSE_REP2") // DataSource is not exposed
+    public JDBCVectorStoreRecordCollection(
+        @Nonnull DataSource dataSource,
+        @Nonnull String collectionName,
+        @Nonnull JDBCVectorStoreRecordCollectionOptions<Record> options) {
+        this.collectionName = collectionName;
+        this.options = options;
+
+        // If record definition is not provided, create one from the record class
+        recordDefinition = options.getRecordDefinition() == null
+            ? VectorStoreRecordDefinition.fromRecordClass(options.getRecordClass())
+            : options.getRecordDefinition();
+
+        // If mapper is not provided, set a default one
+        if (options.getVectorStoreRecordMapper() == null) {
+            vectorStoreRecordMapper = JDBCVectorStoreRecordMapper.<Record>builder()
+                .withRecordClass(options.getRecordClass())
+                .withVectorStoreRecordDefinition(recordDefinition)
+                .build();
+        } else {
+            vectorStoreRecordMapper = options.getVectorStoreRecordMapper();
+        }
+
+        // If the query provider is not provided, set a default one
+        if (options.getQueryProvider() == null) {
+            this.queryProvider = JDBCVectorStoreDefaultQueryProvider.builder()
+                .withDataSource(dataSource)
+                .build();
+        } else {
+            this.queryProvider = options.getQueryProvider();
+        }
+
+        // Check if the types are supported
+        queryProvider.validateSupportedTypes(options.getRecordClass(), recordDefinition);
+    }
+
+    /**
+     * Gets the name of the collection.
+     *
+     * @return The name of the collection.
+     */
+    @Override
+    public String getCollectionName() {
+        return collectionName;
+    }
+
+    /**
+     * Checks if the collection exists in the store.
+     *
+     * @return A Mono emitting a boolean indicating if the collection exists.
+     */
+    @Override
+    public Mono<Boolean> collectionExistsAsync() {
+        return Mono.fromCallable(
+            () -> queryProvider.collectionExists(this.collectionName))
+            .subscribeOn(Schedulers.boundedElastic());
+    }
+
+    /**
+     * Creates the collection in the store.
+     *
+     * @return A Mono representing the completion of the creation operation.
+     */
+    @Override
+    public Mono<Void> createCollectionAsync() {
+        return Mono.fromRunnable(
+            () -> queryProvider.createCollection(this.collectionName, options.getRecordClass(),
+                recordDefinition))
+            .subscribeOn(Schedulers.boundedElastic())
+            .then();
+    }
+
+    /**
+     * Creates the collection in the store if it does not exist.
+     *
+     * @return A Mono representing the completion of the creation operation.
+     */
+    @Override
+    public Mono<Void> createCollectionIfNotExistsAsync() {
```

Comment:
Should these CRUD methods return some sort of status? Or should the user expect some sort of SKException if the operation fails? The javadoc should state if SKException is thrown unchecked.